packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -3118,9 +3118,7 @@ class SemanticsOwner extends ChangeNotifier {
       final CustomSemanticsAction action = CustomSemanticsAction.getAction(actionId)!;
       builder.updateCustomAction(id: actionId, label: action.label, hint: action.hint, overrideId: action.action?.index ?? -1);
     }
-    // TODO(a-wallen): https://github.com/flutter/flutter/issues/112221
-    // ignore: deprecated_member_use
-    SemanticsBinding.instance.platformDispatcher.updateSemantics(builder.build());
+    SemanticsBinding.instance.platformDispatcher.views.first.updateSemantics(builder.build());
```

Comment:
Instead of assuming a single view here, one should be able to configure the SemanticsOwner with the view it should use. I would actually suggest to make the SemanticsOwner configurable with a `onSemanticsUpdate` callback that is invoked with `builder.build()` as argument right here. (Ultimately, it should probably be the RendererBinding providing that callback as it also configures the FlutterView into which the visuals are rendered.)

Reply:
@goderbauer https://github.com/flutter/flutter/pull/113382/commits/32eec8dfa5f02d95ce6976238c9b38623974dcfb is my interpretation of

> I would actually suggest to make the SemanticsOwner configurable with a onSemanticsUpdate callback that is invoked with builder.build() as argument right here. (Ultimately, it should probably be the RendererBinding providing that callback as it also configures the FlutterView into which the visuals are rendered.)

can you confirm that this is what https://github.com/flutter/flutter/pull/113382/commits/32eec8dfa5f02d95ce6976238c9b38623974dcfb is what you mean.

Reply:
Yeah that is looking good.